src/media/js/addon/actions/review.js

```diff
@@ -43,42 +55,68 @@ export function fetch() {
 }
 
 
-export function publish(slug) {
+export function publish(addonSlug, versionId) {
+  /* Publish version of an add-on. */
   return (dispatch, getState) => {
     const apiArgs = getState().apiArgs || {};
     const publishUrl = Url(
-      urlJoin(process.env.MKT_API_ROOT, `extensions/queue/${slug}/publish/`)
+      urlJoin(
+        process.env.MKT_API_ROOT,
+        `extensions/extension/${addonSlug}/versions/${versionId}/publish/`
+      )
     ).q(apiArgs);
 
-    dispatch(publishPending(slug));
+    dispatch(publishPending({
+      addonSlug,
+      versionId
+    }));
     req
       .post(publishUrl)
       .then((res, err) => {
         if (res.status >= 400) {
-          dispatch(publishError(slug));
+          dispatch(publishError({
```

Comment:
this is actually wrong, it should be two functions, one for resolve and one for reject where the reject takes the error object.

I'm also still waiting on my pr to redux-actions to do FSA-errors by simply passing an error object as the payload using the same action creator